components/cf-app-push/frontend/src/view/deploy-app-workflow/deploy-app-bits.html

```diff
@@ -24,69 +24,124 @@
+                    </div>
+                    <div>
+                      <span translate>deploy-app-dialog.step-info.github.branch-last-commit-date</span> {{ step.data.githubCommit.commit.author.date | momentDateFormat }}
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
         </div>
-        {{ step.data.githubProject.description | limitTo:50}} <span ng-if="step.data.githubProject.description.length > 50">...</span>
       </div>
     </div>
 
-    <div ng-if="step.data.githubCommit" class="lastCommit">
-      <div ng-show="step.data.githubCommit.author.avatar_url" class="committerAvatar">
-        <img ng-src="{{step.data.githubCommit.author.avatar_url}}"/>
-      </div>
-      <div class="lastCommit-info">
-        <div>
-          <span translatE>deploy-app-dialog.step-info.github.branch-last-commit</span>
-          <a href="https://www.github.com/{{step.userInput.githubProject}}/commit/{{step.data.githubCommit.sha}}"
-             rel="noopener noreferrer" target="_blank">{{ step.data.githubCommit.sha | limitTo:6}} {{ step.data.githubCommit.commit.message | limitTo:50}}<span ng-if="step.data.githubCommit.commit.message.length > 50">...</span>
-          </a>
-        </div>
-        <div>
-          <span translate>deploy-app-dialog.step-info.github.branch-last-commit-date</span> {{ step.data.githubCommit.commit.author.date | momentDateFormat }}
-        </div>
-        <div class="lastCommitter">
-          <span translate>deploy-app-dialog.step-info.github.branch-last-commit-committer</span>{{ step.data.githubCommit.author.login}}
+    <div class="deploy-app-source-local">
+      <radio-input ng-if="!step.folderSupport" ng-model="step.userInput.sourceType" input-value="'local'" input-label="deploy-app-dialog.step-info.local.file.label"></radio-input>
+      <radio-input ng-if="step.folderSupport" ng-model="step.userInput.sourceType" input-value="'local'" input-label="deploy-app-dialog.step-info.local.folder.label"></radio-input>
+      <div class="deploy-app-github">
+        <div class="deploy-app-github-form">
+          <div class="form-group form-group-large deploy-app-file-controls" focusable-input>
+            <label ng-if="!step.folderSupport" class="control-label" translate>deploy-app-dialog.step-info.local.file.field</label>
+            <label ng-if="step.folderSupport" class="control-label" translate>deploy-app-dialog.step-info.local.folder.field</label>
+            <input fileread="step.userInput.localPathFile" type="file" class="form-control file-chooser" name="localPathSelectFile" id="localPathSelectFile"></input>
+            <input webkitdirectory fileread="step.userInput.localPathFile" type="file" class="form-control file-chooser" name="localPathSelectFolder" id="localPathSelectFolder"></input>
```

Comment:
I left these not disabled, so that the user can click to select without having to first click the radio button.

I guess this is going to get refactored again, anyway, so I'd suggest we punt for now.